src/core/tests/util/graph_comparator.hpp

```diff
@@ -737,6 +741,39 @@ struct Equal<std::shared_ptr<Constant>> {
         return false;
     }
 };
+
+template <>
+struct Equal<std::shared_ptr<ov::Dimension>> {
+    static bool equal_value(const std::shared_ptr<ov::Dimension>& dim1, const std::shared_ptr<ov::Dimension>& dim2) {
+        if (dim1->is_dynamic() != dim2->is_dynamic())
+            return false;
+        if (dim1->get_min_length() != dim2->get_min_length())
+            return false;
+        return (dim1->get_max_length() == dim2->get_max_length());
+    }
+};
+
+template <>
+struct Equal<std::shared_ptr<ov::PartialShape>> {
+    static bool equal_value(const std::shared_ptr<ov::PartialShape>& shape1,
+                            const std::shared_ptr<ov::PartialShape>& shape2) {
+        if (shape1->rank().is_dynamic() != shape2->rank().is_dynamic())
+            return false;
+        if (shape1->rank().is_dynamic())
+            return true;
+        if (shape1->size() != shape2->size())
+            return false;
+
+        for (auto i = 0; i < shape1->size(); i++) {
+            const auto dim1 = std::make_shared<ov::Dimension>((*shape1)[i]);
+            const auto dim2 = std::make_shared<ov::Dimension>((*shape2)[i]);
+            if (!equal::Equal<std::shared_ptr<ov::Dimension>>::equal_value(dim1, dim2))
+                return false;
+        }
+        return true;
```

Comment:
We don't have to invent new comparators for classes which already have them.
https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/core/partial_shape.hpp#L179

Reply:
Used operator== in this method.